packages/docs/src/pages/guides/jsx-pragma.mdx

```diff
@@ -55,3 +64,120 @@ The preferred way of using these functions is by adding the custom pragma commen
 See the [`sx` prop][] docs to learn more.
 
 [`sx` prop]: /sx-prop
+
+## Automatic JSX runtime
+
+React v17 introduced a
+[new JSX transform](https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html),
+also called "automatic runtime" (backported to React v16.14.0) If you use any of
+those versions together with a configurable transpiler (such as Babel or
+TypeScript), you can configure JSX to use the automatic runtime (globally, for
+your entire app), and no longer need to use the custom pragma comments in your
+files to use `sx`.
+
+Here are a few examples showing how to do that depending how you build your
+application:
+
+### Using @babel/preset-react
+
+```js
+// babel.config.js
+module.exports = {
+  presets: [
+    [
+      '@babel/preset-react',
+      {
+        importSource: 'theme-ui', // or '@theme-ui/core'
+        runtime: 'automatic',
+        throwIfNamespace: false,
+      },
+    ],
+  ],
+  // ...
+}
+```
+
+NOTE: this requires
+[babel >= 7.9.0](https://babeljs.io/docs/en/babel-preset-react#react-automatic-runtime)
+
+### Using @babel/plugin-transform-react-jsx
+
+This plugin is included in the preset above, but can also used as a standalone
+babel plugin.
+
+```js
+// babel.config.js
+module.exports = {
+  presets: [
+    [
+      '@babel/plugin-transform-react-jsx',
+      {
+        importSource: 'theme-ui', // or '@theme-ui/core'
+        runtime: 'automatic',
+        throwIfNamespace: false,
+      },
+    ],
+  ],
+  // ...
+}
+```
+
+NOTE: this requires
+[babel >= 7.9.0](https://babeljs.io/docs/en/babel-plugin-transform-react-jsx#react-automatic-runtime)
+
+### Using TypeScript
+
+If you use TypeScript to transpile your source code with `tsc` (or only
+typecheck), or for instance to run tests with `ts-jest`
+
+```json
+// tsconfig.json
+{
+  "compilerOptions": {
+    "jsx": "react-jsx",
+    "jsxImportSource": "theme-ui"
+  }
+  // ...
+}
+```
+
+NOTE: this requires
+[TypeScript >= 4.1](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-1.html#react-17-jsx-factories)
+
+NOTE: in order to typecheck `sx` using automatic runtime, you will also need to
```

Comment:
Is this also the case when `"jsxImportSource": "theme-ui"` is configured in TypeScript config?

Reply:
Huuuh, I think this is above my TS understanding to be honest, but I can re-try in a project 😁

Reply:
I was under impression that TS correctly found types from configured importSource, but I'll have to double-check.

Reply:
Seems like you're right actually, good catch :)
I will remove it

Reply:
@hasparus just figured out why I thought this was still necessary:

If you apply `sx` to a non theme-ui component without manually providing `sx` types, then it does not typecheck even with `jsxImportSource: "theme-ui"` inside tsconfig.json (even though sx gets correctly compiled)

```tsx
// StyledForm.tsx
import { Button, Flex, Input, Label } from 'theme-ui';

// Some interface not explicitly typing "sx"
export interface StyledFormProps {
  onChange(data: Record<string, unknown>): void;
}

const StyledForm: React.FC<StyledFormProps> = ({
  onChange,
  ...props
}) => (
  <form {...props}> // spreading props, including sx --> Emotion correctly compiles it to a className
    <Flex sx={{ flexFlow: 'column', p: 's' }}>
      <Label htmlFor="input.test">Test</Label>
      <Input
        id="input.test"
        type="text"
        onChange={
          (event) => onChange({
            test: event.target.value
          })
        }
      />
    </Flex>
  </form>
);

export default StyledForm;
```

TS error when forwarding `sx`

```tsx
// App.tsx

const App = () => (
  <StyledForm
    onChange={console.log}
    sx={{
      // Type '{ /* ... */ sx: { ...; }; }' is not assignable to type 'IntrinsicAttributes & StyledFormProps & { children?: ReactNode; }'.
      //   Property 'sx' does not exist on type 'IntrinsicAttributes & StyledFormProps & { children?: ReactNode; }'.
      'px': 'm'
    }}
  />
);
```

This can be easily fixed by adding `sx?: ThemeUIStyleObject` to `StyledProps` in that case, just slightly sad that we get a TS error even though the className works fine IMHO, but I don't mind adding the types wherever it is needed

Could/Should the automatic `sx` typings be added to all JSX components instead of only theme-ui ones? 🤔

Reply:
Hey @flo-sch 👋 @hasparus doesn't have access to his computer, and he asked me to write this 😄 

Theme UI types add `sx` prop only to components which props already accept `className`, as `sx` transpiles to `className`. 

The code responsible for this is at [packages/core/src/jsx-namespace.ts#L3](https://github.com/system-ui/theme-ui/blob/develop/packages/core/src/jsx-namespace.ts#L3).

Here's a CodeSandbox, so you can try it out: https://codesandbox.io/s/frosty-https-wpp7f?file=/src/index.tsx

Reply:
Hi @beerose, nice to meet you and thanks for relaying the answer 😁
(not sure if the sandbox link was the correct one though, seems more about GraphQL in there?) 

That make sense in term of typings I guess, I did reproduce it (even without automatic runtime):
https://codesandbox.io/s/sx-props-typings-automatic-runtime-dte1v?file=/src/App.tsx

My only claim was that, since Emotion successfully compiles it anyway, it felt like a "false-negative" from TS to complain on missing `sx` types in that case (in that App.tsx, there is a TS complain on the first `<Form sx={{ }} />` since it does not add types for className or sx)

But I guess this is another discussion here, not related to automatic runtime in itself

Reply:
Hey! I'm not sure how it happened but yes, the link was wrong 🤦‍♀️
Here's the valid one: https://codesandbox.io/s/affectionate-architecture-utrwm?file=/src/index.tsx
Sorry!

Reply:
> My only claim was that, since Emotion successfully compiles it anyway, it felt like a "false-negative" from TS to complain on missing sx types in that case (in that App.tsx, there is a TS complain on the first <Form sx={{ }} /> since it does not add types for className or sx)

IMHO this isn't a false negative — `sx` is compiled to `className` which is not handled so we see almost the same error as if we passed a `className`, meaning "this component cannot be styled from the outside".